src/diagnostics.rs

```diff
@@ -24,6 +19,14 @@ pub struct Diagnostics {
     writer: RefCell<Box<dyn WriteColor>>,
 }
 
+const ENCODING_ERROR_MAX_CONTEXT_LENGTH: usize = 80;
```

Comment:
my usual nitpick

Reply:
sorry. I forgot the WIP on the pull request name. diagnostics.rs is not yet rewritten. Most of the file is just commented out to make it compile.